[Issue] buildAddonStyles  (implicit-styles)

```js
Build Error (WaitForTrees)
Cannot find module './@linkedin/artdeco-button.css' from '/private/tmp/embroider-workspace/node_modules/@linkedin/artdeco-button'
```

Specifically: https://github.com/embroider-build/embroider/blob/118c7bf10fff8e92d15d1ef31ac2c05b74146150/packages/core/src/app.ts#L195

Is attempting to use the node resolution algorithm, but the intended staged target of `@linkedin/artdeco-button` residing in `/private/tmp/embroider-workspace/node_modules/@linkedin/@artdeco-button` is not resolvable, as it has no resolvable main.

As it's `package.json` `main` property is empty, the resolution algorithm will attempt to tr3eat `@linkedin/artdeco-button/index.js` as it's implicit main, but as that file is not present, the resolution algorithm assumes a miss and moves on.

It is worth pointing out that `@linkedin/artdeco-button/index.scss` does exist, but that file is not considered by the node resolution algorithm.

I think i have a vague idea of what your goal is here, but it isn't sufficiently crisp to suggest a path forward. We should likely sync up.



<details>
  <summary>error.dump.log</summary>

```js
=================================================================================

ENV Summary:

  TIME: Sun Feb 24 2019 12:55:25 GMT-0800 (Pacific Standard Time)
  TITLE: ember
  ARGV:
  - /Users/spenner/.gradle/pemberly/node-v10.6.0-darwin-x64/bin/node
  - /Users/spenner/.config/node/default/bin/ember
  - b
  EXEC_PATH: /Users/spenner/.gradle/pemberly/node-v10.6.0-darwin-x64/bin/node
  TMPDIR: /var/folders/4r/whc65vwj1xggvvky3yy1cp9m000mw4/T
  SHELL: /usr/local/bin/fish
  PATH:
  - /Users/spenner/.gradle/pemberly/yarnpkg-1.13.0/bin
  - /Users/spenner/.gradle/pemberly/node-v10.6.0-darwin-x64/bin
  - /Users/spenner/l/pemberly-example-web_trunk/.gradle/pemberly
  - /Users/spenner/.cargo/bin
  - /Users/spenner/.fzf/bin
  - /Users/spenner/.local/bin
  - /Applications/Racket v7.1/bin/
  - /Users/spenner/.config/node/default/bin
  - /usr/local/bin
  - /usr/bin
  - /bin
  - /usr/sbin
  - /sbin
  - /usr/local/linkedin/bin
  - /Applications/VMware
  - Fusion.app/Contents/Public
  - /export/content/linkedin/bin
  - /Applications/Wireshark.app/Contents/MacOS
  - /Users/spenner/.local/bin
  - /Applications/Racket v7.1/bin/
  - /Users/spenner/.config/node/default/bin
  PLATFORM: darwin x64
  FREEMEM: 2144165888
  TOTALMEM: 17179869184
  UPTIME: 1119779
  LOADAVG: 2.14306640625,1.88232421875,1.93408203125
  CPUS:
  - Intel(R) Core(TM) i7-7920HQ CPU @ 3.10GHz - 3100
  - Intel(R) Core(TM) i7-7920HQ CPU @ 3.10GHz - 3100
  - Intel(R) Core(TM) i7-7920HQ CPU @ 3.10GHz - 3100
  - Intel(R) Core(TM) i7-7920HQ CPU @ 3.10GHz - 3100
  - Intel(R) Core(TM) i7-7920HQ CPU @ 3.10GHz - 3100
  - Intel(R) Core(TM) i7-7920HQ CPU @ 3.10GHz - 3100
  - Intel(R) Core(TM) i7-7920HQ CPU @ 3.10GHz - 3100
  - Intel(R) Core(TM) i7-7920HQ CPU @ 3.10GHz - 3100
  ENDIANNESS: LE
  VERSIONS:
  - ares: 1.14.0
  - cldr: 33.0
  - http_parser: 2.8.0
  - icu: 61.1
  - modules: 64
  - napi: 3
  - nghttp2: 1.32.0
  - node: 10.6.0
  - openssl: 1.1.0h
  - tz: 2018c
  - unicode: 10.0
  - uv: 1.21.0
  - v8: 6.7.288.46-node.13
  - zlib: 1.2.11

ERROR Summary:

  - broccoliBuilderErrorStack: Error: Cannot find module './@linkedin/artdeco-button.css' from '/private/tmp/embroider-workspace/node_modules/@linkedin/artdeco-button'
    at Function.module.exports [as sync] (/Users/spenner/src/embroider-build/embroider/node_modules/resolve/lib/sync.js:58:15)
    at AppBuilder.impliedAddonAssets (/Users/spenner/src/embroider-build/embroider/packages/core/src/app.js:109:51)
    at AppBuilder.impliedAssets (/Users/spenner/src/embroider-build/embroider/packages/core/src/app.js:84:27)
    at AppBuilder.insertEmberApp (/Users/spenner/src/embroider-build/embroider/packages/core/src/app.js:150:35)
    at AppBuilder.prepareAsset (/Users/spenner/src/embroider-build/embroider/packages/core/src/app.js:196:18)
    at AppBuilder.prepareAssets (/Users/spenner/src/embroider-build/embroider/packages/core/src/app.js:206:18)
    at AppBuilder.updateAssets (/Users/spenner/src/embroider-build/embroider/packages/core/src/app.js:270:27)
    at AppBuilder.build (/Users/spenner/src/embroider-build/embroider/packages/core/src/app.js:324:38)
    at WaitForTrees.wait_for_trees_1.default [as buildHook] (/Users/spenner/src/embroider-build/embroider/packages/core/src/build-stage.js:37:31)
    at process._tickCallback (internal/process/next_tick.js:68:7)
  - code: [undefined]
  - codeFrame: Cannot find module './@linkedin/artdeco-button.css' from '/private/tmp/embroider-workspace/node_modules/@linkedin/artdeco-button'
  - errorMessage: Cannot find module './@linkedin/artdeco-button.css' from '/private/tmp/embroider-workspace/node_modules/@linkedin/artdeco-button'
        at WaitForTrees (@embroider/compat/app)
-~- created here: -~-
    at new Plugin (/Users/spenner/src/embroider-build/embroider/node_modules/broccoli-plugin/index.js:7:31)
    at new WaitForTrees (/Users/spenner/src/embroider-build/embroider/packages/core/src/wait-for-trees.js:30:9)
    at CompatApp.get tree (/Users/spenner/src/embroider-build/embroider/packages/core/src/build-stage.js:26:16)
    at CompatApp.<anonymous> (/Users/spenner/src/embroider-build/embroider/node_modules/typescript-memoize/dist/memoize-decorator.js:67:52)
    at new PackagerRunner (/Users/spenner/src/embroider-build/embroider/packages/core/src/to-broccoli-plugin.js:11:26)
    at Object.defaultPipeline [as compatBuild] (/Users/spenner/src/embroider-build/embroider/packages/compat/src/default-pipeline.js:28:12)
    at module.exports (/Users/spenner/l/pemberly-example-web_trunk/ember-cli-build.js:57:39)
    at Builder.readBuildFile (/Users/spenner/l/pemberly-example-web_trunk/node_modules/ember-cli/lib/models/builder.js:52:14)
    at Builder.setupBroccoliBuilder (/Users/spenner/l/pemberly-example-web_trunk/node_modules/ember-cli/lib/models/builder.js:66:22)
    at new Builder (/Users/spenner/l/pemberly-example-web_trunk/node_modules/ember-cli/lib/models/builder.js:32:10)
    at BuildTask.run (/Users/spenner/l/pemberly-example-web_trunk/node_modules/ember-cli/lib/tasks/build.js:15:19)
    at Promise.resolve.then (/Users/spenner/l/pemberly-example-web_trunk/node_modules/ember-cli/lib/models/command.js:243:46)
    at tryCatcher (/Users/spenner/l/pemberly-example-web_trunk/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:323:19)
    at invokeCallback (/Users/spenner/l/pemberly-example-web_trunk/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:495:31)
    at /Users/spenner/l/pemberly-example-web_trunk/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:559:14
    at flush (/Users/spenner/l/pemberly-example-web_trunk/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:2402:5)
    at process._tickCallback (internal/process/next_tick.js:61:11)
-~- (end) -~-
  - errorType: Build Error
  - location:
    - column: [undefined]
    - file: [undefined]
    - line: [undefined]
    - treeDir: [undefined]
  - message: Cannot find module './@linkedin/artdeco-button.css' from '/private/tmp/embroider-workspace/node_modules/@linkedin/artdeco-button'
        at WaitForTrees (@embroider/compat/app)
-~- created here: -~-
    at new Plugin (/Users/spenner/src/embroider-build/embroider/node_modules/broccoli-plugin/index.js:7:31)
    at new WaitForTrees (/Users/spenner/src/embroider-build/embroider/packages/core/src/wait-for-trees.js:30:9)
    at CompatApp.get tree (/Users/spenner/src/embroider-build/embroider/packages/core/src/build-stage.js:26:16)
    at CompatApp.<anonymous> (/Users/spenner/src/embroider-build/embroider/node_modules/typescript-memoize/dist/memoize-decorator.js:67:52)
    at new PackagerRunner (/Users/spenner/src/embroider-build/embroider/packages/core/src/to-broccoli-plugin.js:11:26)
    at Object.defaultPipeline [as compatBuild] (/Users/spenner/src/embroider-build/embroider/packages/compat/src/default-pipeline.js:28:12)
    at module.exports (/Users/spenner/l/pemberly-example-web_trunk/ember-cli-build.js:57:39)
    at Builder.readBuildFile (/Users/spenner/l/pemberly-example-web_trunk/node_modules/ember-cli/lib/models/builder.js:52:14)
    at Builder.setupBroccoliBuilder (/Users/spenner/l/pemberly-example-web_trunk/node_modules/ember-cli/lib/models/builder.js:66:22)
    at new Builder (/Users/spenner/l/pemberly-example-web_trunk/node_modules/ember-cli/lib/models/builder.js:32:10)
    at BuildTask.run (/Users/spenner/l/pemberly-example-web_trunk/node_modules/ember-cli/lib/tasks/build.js:15:19)
    at Promise.resolve.then (/Users/spenner/l/pemberly-example-web_trunk/node_modules/ember-cli/lib/models/command.js:243:46)
    at tryCatcher (/Users/spenner/l/pemberly-example-web_trunk/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:323:19)
    at invokeCallback (/Users/spenner/l/pemberly-example-web_trunk/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:495:31)
    at /Users/spenner/l/pemberly-example-web_trunk/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:559:14
    at flush (/Users/spenner/l/pemberly-example-web_trunk/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:2402:5)
    at process._tickCallback (internal/process/next_tick.js:61:11)
-~- (end) -~-
  - name: BuildError
  - nodeAnnotation: @embroider/compat/app
  - nodeName: WaitForTrees
  - originalErrorMessage: Cannot find module './@linkedin/artdeco-button.css' from '/private/tmp/embroider-workspace/node_modules/@linkedin/artdeco-button'
  - stack: Error: Cannot find module './@linkedin/artdeco-button.css' from '/private/tmp/embroider-workspace/node_modules/@linkedin/artdeco-button'
    at Function.module.exports [as sync] (/Users/spenner/src/embroider-build/embroider/node_modules/resolve/lib/sync.js:58:15)
    at AppBuilder.impliedAddonAssets (/Users/spenner/src/embroider-build/embroider/packages/core/src/app.js:109:51)
    at AppBuilder.impliedAssets (/Users/spenner/src/embroider-build/embroider/packages/core/src/app.js:84:27)
    at AppBuilder.insertEmberApp (/Users/spenner/src/embroider-build/embroider/packages/core/src/app.js:150:35)
    at AppBuilder.prepareAsset (/Users/spenner/src/embroider-build/embroider/packages/core/src/app.js:196:18)
    at AppBuilder.prepareAssets (/Users/spenner/src/embroider-build/embroider/packages/core/src/app.js:206:18)
    at AppBuilder.updateAssets (/Users/spenner/src/embroider-build/embroider/packages/core/src/app.js:270:27)
    at AppBuilder.build (/Users/spenner/src/embroider-build/embroider/packages/core/src/app.js:324:38)
    at WaitForTrees.wait_for_trees_1.default [as buildHook] (/Users/spenner/src/embroider-build/embroider/packages/core/src/build-stage.js:37:31)
    at process._tickCallback (internal/process/next_tick.js:68:7)

=================================================================================
```
</details>

--- a/packages/compat/src/v1-addon.ts
+++ b/packages/compat/src/v1-addon.ts
@@ -3,7 +3,7 @@ import { Memoize } from 'typescript-memoize';
 import { dirname } from 'path';
 import { sync as pkgUpSync } from 'pkg-up';
 import { join } from 'path';
-import { existsSync, pathExistsSync } from 'fs-extra';
+import { existsSync, pathExistsSync, readdirSync } from 'fs-extra';
 import Funnel, { Options as FunnelOptions } from 'broccoli-funnel';
 import { UnwatchedDir } from 'broccoli-source';
 import DependencyAnalyzer from './dependency-analyzer';
@@ -439,11 +439,16 @@ export default class V1Addon implements V1Package {
   private buildAddonStyles(built: IntermediateBuild) {
     let addonStylesTree = this.addonStylesTree();
     if (addonStylesTree) {
-      built.trees.push(addonStylesTree);
-      if (!built.staticMeta['implicit-styles']) {
-        built.staticMeta['implicit-styles'] = [];
-      }
-      built.staticMeta['implicit-styles'].push(`./${this.name}.css`);
+      let discoveredFiles: string[] = [];
+      let tree = new AddToTree(addonStylesTree, outputPath => {
+        discoveredFiles = readdirSync(outputPath).filter(name => name.endsWith('.css'));
+      });
+      built.trees.push(tree);
+      built.dynamicMeta.push(() => {
+        return {
+          'implicit-styles': discoveredFiles.map(f => `./${f}`),
+        };
+      });
     }
   }
 

